hw/ip/pwm/data/pwm.hjson

```diff
@@ -0,0 +1,203 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+{ name: "pwm",
+  clock_primary: "clk_i",
+  other_clock_list: [ "clk_core_i" ],
+  reset_primary: "rst_ni",
+  other_reset_list: [ "rst_core_ni" ],
+  bus_device: "tlul",
+  bus_host: "none",
+  regwidth: "32",
+  param_list: [
+    { name: "NOutputs",
+      desc: "Number of PWM outputs",
+       type: "int",
+      default: "6",
+    }
+  ]
+  available_output_list: [
+    { name:  "pwm"
+      desc:  "pulse output"
+      width: "6"
+    }
+  ]
+  registers: [
+    { name: "REGEN",
+      desc: "Register write enable for all control registers",
+      swaccess: "rw1c",
+      hwaccess: "hro",
+      fields: [
+        { bits: "0",
+          desc: ''' When true, all writable registers can be modified.
+                    When false, they become read-only. Defaults true, write
+                    one to clear. This can be cleared after initial
+                    configuration at boot in order to lock in the listed
+                    register settings.'''
+          resval: 1
+        }
+      ]
+    }
+    { name: "CFG",
+      desc: "Configuration register",
+      swaccess: "rw",
+      fields: [
+        { bits: "31",
+          name: "CNTR_EN",
+          desc: '''Assert this bit to enable the PWM phase counter.
+                   Clearing this bit disables and resets the phase counter.'''
+          resval: "0x0"
+        },
+        { bits: "30:27",
+          name: "DC_RESN"
+          desc: '''Phase Resolution (logarithmic). All duty-cycle and phase
```

Comment:
@martin-lueker it might just be me...but this strikes me as kind of confusing to read (not sure if anyone else has the same feeling).

Does this resolution effectively boil down to how big the step size is every time there is a clock beat? (as defined by the divider below)?  The counter as I understand is always fixed at 16b, so the smaller the value programmed here, the "larger" step you're taking right?

In terms of implementation, is there a clever way to take advantage of this? Or do they end up as counters with pre-selected steps?